### ISSUE TITLE
CastCannotSucceed analyzer

### ISSUE THREAD

**Issue by Hosch250** [Vannevelj/VSDiagnostics#138]
Say I have this:

    if (value is Type)
    {
        var v = (Type) value;
    }

VS knows that the cast will succeed and doesn't display an error.  But, if I do this:

    if (value is Type)
    {
        var v = (Type1) value;
    }

VS will tell me that the cast might fail.  It would be nice if it could tell me whether or not the cast would fail.  It might not be possible, but, this does tell me that the cast will fail:

    var ch = 'r';
    var b = (bool) ch;

And this tells me that it will succeed (or rather, it doesn't give any warning at all, which is the same thing):

    var ch = 'r';
    var b = (int) ch;

Thing is, `char` probably has a conversion operator to `int`, and `char`, `int`, and `bool` are all `struct`s, so that would be how they can tell, while I am thinking of classes.

**Comment by Vannevelj** [comment]
Yeah, it's worth looking further into this. Some relevant notes on the subject:

* Can it be subclassed? (`struct` or `sealed classes`)
* Does it overload the equality operator between the two types?
* Is there an overload available between one or more super types? (I'm not sure if this is possible, so look into it)
